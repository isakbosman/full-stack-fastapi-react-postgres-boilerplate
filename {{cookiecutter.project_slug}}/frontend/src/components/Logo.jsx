import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { ReactComponent as Icon } from 'assets/media/brand/icon.svg';
import { ReactComponent as RRS } from 'assets/media/brand/react-redux-saga.svg';

export const Wrapper = styled.div`
  align-items: flex-start;
  display: inline-flex;
  font-size: 0;

  svg {
    height: 4.2rem;
    max-height: 100%;
    width: auto;
  }
`;

const Logo = ({ type = 'icon' }) => <Wrapper>{type === 'icon' ? <Icon /> : <RRS />}</Wrapper>;

Logo.propTypes = {
  type: PropTypes.string,
};

export default Logo;
