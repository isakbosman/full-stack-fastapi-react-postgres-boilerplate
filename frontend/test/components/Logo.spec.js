import React from 'react';

import Logo from 'components/Logo';

describe('Logo', () => {
  const wrapper = mount(<Logo />);

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
