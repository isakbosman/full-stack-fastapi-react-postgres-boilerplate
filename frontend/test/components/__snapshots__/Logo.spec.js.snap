// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render properly 1`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 0;
}

.c0 svg {
  height: 4.2rem;
  max-height: 100%;
  width: auto;
}

<Wrapper>
  <div
    className="c0"
  >
    SVG
  </div>
</Wrapper>
`;
